src/generic-container/generic-container.test.ts

```diff
@@ -71,6 +71,25 @@ describe("GenericContainer", () => {
     await container.stop();
   });
 
+  it("should wait for a new log after restart", async () => {
+    const start = new Date();
+    const container = await new GenericContainer("cristianrgreco/testcontainer:1.1.14")
+      .withCommand(["/bin/sh", "-c", 'sleep 2; echo "Ready"'])
+      .withWaitStrategy(Wait.forLogMessage("Ready"))
+      .start();
+
+    expect(new Date().getTime() - start.getTime()).toBeGreaterThanOrEqual(2_000);
+
+    // Add a short delay, as otherwise we'll still be too fast and restart the container
+    // on the same second as it previously got ready (which will mess with the log detection)
+    await new Promise((resolve) => setTimeout(resolve, 1_000));
```

Comment:
Why won't the same problem occur in production cases?

Reply:
Because it would assume someone restarts the container on the same second that it was declared ready by the log waiting strategy - this seems highly unlikely, no? I could also manipulate the restarting to force a delay that way. I can't think of any other solution around this, any thoughts?

Reply:
I refactored to not rely on this type of behaviour, it does come with a bit of performance overhead in the log waiting strategy, but perhaps that is not a big deal? Basically, it validates that the log lines it gets are actually after the expected start timestamp.

Reply:
This looks great, I don't think there's any real performance impact here and it lets us be 100% accurate